src/app/script/api/app_script.cpp

```diff
@@ -20,6 +18,23 @@
 #include "app/ui/document_view.h"
 #include "doc/site.h"
 
+#include "script/engine.h"
+#include "script/engine_delegate.h"
+#include "script/script_object.h"
+
+#include <sstream>
+
+class DudScriptObject : public script::InternalScriptObject {
```

Comment:
It's a Dud because it doesn't actually do anything. It's just a placeholder so that the classes can be inspected without having side-effects. (Could also call it a NullScriptObject, but I'm not sure if the Null Object design pattern is that well-known)

> Dud: a thing that fails to work properly or is otherwise unsatisfactory or worthless.